fix interactive "resume --last" prompt handling and reinstate session-id conflict check

dkhokhlov opened the pull request:
### Addressed issues:
- https://github.com/openai/codex/issues/6717#issuecomment-3584958320
- https://github.com/openai/codex/pull/6719#discussion_r2541117561

### Changes
- change `--last` flag to accept optional prompt value (e.g. `--last <PROMPT>`)
- simplify prompt handling by using `last_prompt()` helper method
- reject invalid or nonexistent session IDs with error message instead of silently creating new conversation
- reinstate clap's `conflicts_with` between `--last` and explicit session id check
- add tests for conflict validation, prompt forwarding, and invalid session ID rejection

## Problems

### 1. Interactive `codex resume --last` didn't accept prompt values ([#6717](https://github.com/openai/codex/issues/6717))

**Problem:** The interactive `codex resume --last` command used a boolean flag that couldn't accept a prompt value. Users couldn't provide a prompt directly after `--last`.

**Before:**
```bash
$ codex resume --last "next steps"
# --last was a bool flag, prompt was ignored
```

**After:**
```bash
$ codex resume --last "next steps"
# Now correctly accepts and forwards the prompt
```

### 2. `codex exec resume` silently ignored session_id when used with `--last` ([#6719 discussion](https://github.com/openai/codex/pull/6719#discussion_r2541117561))

**Problem:** When both a session ID and `--last` were provided, the code silently ignored the session ID and used `--last`, potentially resuming the wrong session. This was unsafe for scripts and users.

**Before:**
```bash
$ codex exec resume 123e4567-e89b-12d3-a456-426614174000 --last
# Silently ignored the session_id and resumed most recent session
# Could touch the wrong session unintentionally
```

**After:**
```bash
$ codex exec resume 123e4567-e89b-12d3-a456-426614174000 --last
error: the argument '--last' cannot be used with '<SESSION_ID>'
# Fails fast with clear error message
```

### 3. Invalid session IDs silently created new conversations

**Problem:** When an invalid or nonexistent session ID was provided, `codex exec resume` silently created a new conversation instead of erroring, causing confusion.

**Before:**
```bash
$ codex exec resume sdfsdfgfds2546456 "2+2"
# Silently creates a new conversation (confusing!)
$ codex exec resume 123e4567-e89b-12d3-a456-426614174000 "2+2"
# (UUID format valid but session doesn't exist)
# Silently creates a new conversation
```

**After:**
```bash
$ codex exec resume sdfsdfgfds2546456 "2+2"
No saved session found with ID sdfsdfgfds2546456.
# Exits with error code 1
$ codex exec resume 123e4567-e89b-12d3-a456-426614174000 "2+2"
No saved session found with ID 123e4567-e89b-12d3-a456-426614174000.
# Exits with error code 1
```

### Testing

Added tests for:
- Invalid session ID rejection (non-UUID format)
- Nonexistent valid UUID rejection
- Conflict validation between `--last` and session ID
- Prompt forwarding when using `--last <PROMPT>`

All existing tests pass, and the behavior now matches the interactive `codex resume` command which already had proper error handling.

dkhokhlov committed:
fix resume --last prompt handling and reinstate session-id conflict

- change `--last` flag to accept optional prompt value (e.g. `--last <PROMPT>`)
- simplify prompt handling by using `last_prompt()` helper method
- reject invalid or nonexistent session IDs with error message instead of silently creating new conversation
- reinstate clap's `conflicts_with` between `--last` and explicit session id
- add tests for conflict validation, prompt forwarding, and invalid session ID rejection

etraut-openai commented:
@dkhokhlov, if you're still interested in pursuing this PR, please resolve merge conflicts, take a look at the codex code review comment, and remove the "draft" status once you think it's ready for review by a codex team member. Thanks!

dkhokhlov commented:
> @dkhokhlov, if you're still interested in pursuing this PR, please resolve merge conflicts, take a look at the codex code review comment, and remove the "draft" status once you think it's ready for review by a codex team member. Thanks!

on it

dkhokhlov committed:
fix: remove duplicate test-util feature in tui Cargo.toml

  - Remove redundant "test-util" entry from tokio features list
  - Keep single test-util feature as intended
  - Fix merge conflict artifact from previous rebase

dkhokhlov committed:
fix: review command regression and add test coverage

  - Fix review command breaking due to global prompt validation
  - Revert to command-specific prompt handling for proper separation
  - Keep new --last prompt functionality in resume command
  - Add comprehensive tests for review command without prompts
  - Ensure review commands (--uncommitted, --base, --commit) work as expected

  Fixes P1 regression where "codex exec review" required prompts incorrectly.

dkhokhlov committed:
fix cargo fmt --check

dkhokhlov committed:
Remove review.rs integration tests that made real API calls

- Manual verification confirms regression fix works correctly
